client/src/Try/Loader.purs

```diff
@@ -65,23 +68,39 @@ parseDeps current = Array.mapMaybe go <<< String.split (Pattern "\n") <<< unwrap
   go line = do
     match <- Regex.match requireRegex line
     requirePath <- join $ NonEmpty.index match 1
-    pure $ case resolvePath current requirePath of
-      Just path ->
-        { name: path
-        , path: String.stripPrefix (Pattern "/") path
-        }
-      _ ->
-        { name: requirePath
-        , path: Nothing
-        }
-
-newtype Loader = Loader (JS -> ExceptT String (ContT Unit Effect) (Object JS))
-
-runLoader :: Loader -> JS -> ExceptT String (ContT Unit Effect) (Object JS)
-runLoader (Loader k) = k
-
-makeLoader :: String -> Loader
-makeLoader rootPath = Loader (go Object.empty <<< parseDeps "<file>")
+    pure
+      $ case resolvePath current requirePath of
+          Just path ->
+            { name: path
+            , path: String.stripPrefix (Pattern "/") path
+            }
+          _ ->
+            { name: requirePath
+            , path: Nothing
+            }
+
+{-
+Notes
+
+Could change error handling, but kinda nice to
+just throw the errors from JS.
+
+Assuming makeLoader runLoader pattern is to save
+cache between calls to runLoader.
+
+-}
+newtype Loader
+  = Loader (JS -> Aff (Object JS))
+
+runLoader :: Loader -> JS -> Aff (Object JS)
+runLoader (Loader k) js = do
+  -- Run loader to collect all dependencies for compiled code
+  obj <- k js
+  -- Return dependencies along with compiled code
+  pure $ Object.insert "<file>" js obj
+
+makeLoader :: Loader
+makeLoader = Loader (go Object.empty <<< parseDeps "<file>")
   where
   moduleCache :: Ref (Object Module)
   moduleCache = unsafePerformEffect (Ref.new Object.empty)
```

Comment:
I was hoping there would be a nice way to do this with [memoize](https://pursuit.purescript.org/packages/purescript-memoize/5.0.0), but I couldn't figure it out. It doesn't seem like memoize can be used to bypass repeating effects (such as http requests in this case).